### PR TITLE
ddl: fix the result of executing "desc table" after altering "primary key"

### PR DESCRIPTION
**Pull request by zimulala**
### What problem does this PR solve? 
```
create table t (a int, b int);
alter table t add primary key (a);
desc t;
```
before this PR:
```
+-------+---------+------+------+---------+-------+
| Field | Type    | Null | Key  | Default | Extra |
+-------+---------+------+------+---------+-------+
| a     | int(11) | NO   | UNI  | NULL    |       |
| b     | int(11) | YES  |      | NULL    |       |
+-------+---------+------+------+---------+-------+
```
after this PR:
```
+-------+---------+------+------+---------+-------+
| Field | Type    | Null | Key  | Default | Extra |
+-------+---------+------+------+---------+-------+
| a     | int(11) | NO   | PRI  | NULL    |       |
| b     | int(11) | YES  |      | NULL    |       |
+-------+---------+------+------+---------+-------+
```

### Check List 

Tests 

 - Unit test

Related changes

 - Need to cherry-pick to the release branch

Release note

 - Fix the result of executing "desc table" after altering "primary key"

### CODE REVIEW

File: ddl/db_test.go [hunk 1]
```diff
@@ -1405,6 +1405,14 @@ func (s *testDBSuite5) TestAlterPrimaryKey(c *C) {
 	s.tk.MustExec("alter table test_add_pk add primary key idx(e)")
 	s.tk.MustExec("alter table test_add_pk drop primary key")
 
+	// for describing table
```
Comment (AilinKid) [inline ddl/db_test.go]:
```suggestion
	// Test for describing table.
```

**Comment by cyliu0** [comment]
Still got wrong output with this PR when the primary key is multiple columns.
```
mysql> create table t (a int, b int);
Query OK, 0 rows affected (0.01 sec)

mysql> alter table t add primary key pk(a, b);
Query OK, 0 rows affected (0.03 sec)

mysql> desc t;
+-------+---------+------+------+---------+-------+
| Field | Type    | Null | Key  | Default | Extra |
+-------+---------+------+------+---------+-------+
| a     | int(11) | NO   | PRI  | NULL    |       |
| b     | int(11) | NO   |      | NULL    |       |
+-------+---------+------+------+---------+-------+
2 rows in set (0.00 sec)
```

**Comment by djshow832** [comment]
Any release notes？

File: ddl/db_test.go (hunk 1 above)
Comment (zimulala) [reply ddl/db_test.go]: I think it's OK.

File: ddl/index.go [hunk 2]
```diff
@@ -214,6 +214,12 @@ func buildIndexInfo(tblInfo *model.TableInfo, indexName model.CIStr, idxColNames
 func addIndexColumnFlag(tblInfo *model.TableInfo, indexInfo *model.IndexInfo) {
 	col := indexInfo.Columns[0]
```
Comment (bb7133) [inline ddl/index.go]: Please move this line after `if indexInfo.Primary {...}` block.

**Comment by zimulala** [comment]
Good catch! Done. PTAL @cyliu0

File: ddl/index.go [hunk 3]
```diff
@@ -224,7 +230,11 @@ func addIndexColumnFlag(tblInfo *model.TableInfo, indexInfo *model.IndexInfo) {
 func dropIndexColumnFlag(tblInfo *model.TableInfo, indexInfo *model.IndexInfo) {
 	col := indexInfo.Columns[0]
```
Comment (bb7133) [inline ddl/index.go]: Ditto

**Pull request merged**